sklearn/neighbors/base.py

```diff
@@ -379,11 +379,20 @@ def kneighbors_graph(self, X, n_neighbors=None,
         --------
         NearestNeighbors.radius_neighbors_graph
         """
-        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
-
         if n_neighbors is None:
             n_neighbors = self.n_neighbors
 
+        if X is None:
+            new_graph = False
```

Comment:
Any specific reason you want me to set `n_neighbors += 1` and then delete the diagonal? I have to do the `if-else` in that case too to set the diagonal elements to zero.

I'm also not sure about refactoring the implementation regardless of mode, without setting `return_distance=True` in both cases? I would need an if-else here also to unpack the tuple.